bundles/org.eclipse.jface/src/org/eclipse/jface/dialogs/ScopeChangedEvent.java

```diff
@@ -0,0 +1,68 @@
+/*******************************************************************************
+ * Copyright (c) 2025 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *     Red Hat Inc. - copied and modified from PageChangedEvent.java
+ *******************************************************************************/
+package org.eclipse.jface.dialogs;
+
+import java.util.EventObject;
+
+/**
+ * Event object describing a page selection change. The source of these events
+ * is a page change provider.
+ *
+ * @see IPageChangeProvider
+ * @see IPageChangedListener
+ *
+ * @since 3.38
+ */
+public class ScopeChangedEvent extends EventObject {
+
+	private static final long serialVersionUID = -2652600407410991930L;
+
+	/**
+	 * The changed scope.
+	 */
+	protected int scope;
```

Comment:
If there is a getter and there is no need (?) to set the scope later, I suggest...
```suggestion
	private final int scope;
```